cdap-app-fabric/src/test/java/io/cdap/cdap/internal/TempFolder.java

```diff
@@ -17,18 +17,22 @@
 package io.cdap.cdap.internal;
 
 import com.google.common.base.Throwables;
+import io.cdap.cdap.common.utils.DirUtils;
+import org.junit.rules.TemporaryFolder;
 
 import java.io.File;
 import java.io.IOException;
 
 /**
- * Utility for creating temp folder in unit-tests. All files and dirs created using it will be cleaned up by
- * {@link java.io.File#deleteOnExit()} which is set for every created dir and file automatically.
+ * Utility for creating temp folder in unit-tests.
  *
  * Note: it is preferable to use {@link org.junit.rules.TemporaryFolder} when you can instead of this tool. Use this one
  *       when you don't have access to the unit-test test lifecycle (like static init of some utility classes, etc.).
+ *
+ * @deprecated Don't use this. Use {@link TemporaryFolder} instead
  */
-public class TempFolder {
+@Deprecated
+final class TempFolder {
```

Comment:
Can we just switch to `org.junit.rules.TemporaryFolder` altogether? It's the last place it used and It looks like it can be replaced with class rule pretty easily

Reply:
We can’t easily switch due to some bad static method usage pattern. I would rather do it in a separate PR